tests/descriptor_tests/test_io_descriptors.py

```diff
@@ -147,3 +144,48 @@ def test_cache_locations(self):
             ]
         )
 
+    def test_download_receipt(self):
+        """
+        Tests the download receipt logic
+        """
+
+        sg = self.tk.shotgun
+        root = os.path.join(self.project_root, "cache_root")
+
+        d = sgtk.descriptor.create_descriptor(
+            sg,
+            sgtk.descriptor.Descriptor.APP,
+            {"type": "app_store", "version": "v1.1.1", "name": "tk-bundle"},
+            bundle_cache_root_override=root
+        )
+
+        self.assertEqual(d.get_path(), None)
+        self.assertEqual(d.find_latest_cached_version(), None)
+
+        app_path = os.path.join(root, "app_store", "tk-bundle", "v1.1.1")
+        path = os.path.join(app_path, "info.yml")
+
+        os.makedirs(app_path)
+        fh = open(path, "wt")
+        fh.write("test data\n")
+        fh.close()
+
+        self.assertEqual(d.get_path(), app_path)
```

Comment:
You can probably use the `setUp` to create all the descriptors an path variables you need for both tests.